src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,55 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier randomColor = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        Name[] names = Name.values();
+        int index = random.nextInt(names.length);
+        if (names[index].toString().equals("SQUARE")) {
+            return new Square(
+                     random.nextDouble(),
+                    randomColor.getRandomColor(),
+                    names[index].toString()
+            );
+        }
+        if (names[index].toString().equals("RECTANGLE")) {
+            return new Rectangle(
+                          random.nextDouble(),
+                          random.nextDouble(),
+                          randomColor.getRandomColor(),
+                          names[index].toString()
+                      );
+        }
+        if (names[index].toString().equals("RIGHT_TRIANGLE")) {
+            return new RightTriangle(
+                    random.nextDouble(),
+                    random.nextDouble(),
+                    randomColor.getRandomColor(),
+                    names[index].toString()
+            );
+        }
+        if (names[index].toString().equals("CIRCLE")) {
+            return new Circle(
+                    random.nextDouble(),
+                    randomColor.getRandomColor(),
+                    names[index].toString()
+            );
+        } else {
+            return new IsoscelesTrapezoid(
+                    random.nextDouble(),
+                    random.nextDouble(),
+                    random.nextDouble(),
+                    randomColor.getRandomColor(),
+                    names[index].toString()
+            );
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(10,"WHITE","CIRCLE");
+    }

```

Comment:
You can pass random values to the constructor of a figure instead of generating them inside figure classes. Let's generate random values in `FigureSupplier` and then pass them to the constructors.